docs/reference/specifications/providers.md

```diff
@@ -280,6 +280,7 @@ Below are the supported configuration parameters (note that not all apply to bot
 | cache                 | FLAGD_CACHE                    | enable cache of static flags                                           | String - `lru`, `disabled`   | lru                           | rpc                     |
 | maxCacheSize          | FLAGD_MAX_CACHE_SIZE           | max size of static flag cache                                          | int                          | 1000                          | rpc                     |
 | selector              | FLAGD_SOURCE_SELECTOR          | selects a single sync source to retrieve flags from only that source   | string                       | null                          | in-process              |
+| providerID            | FLAGD_PROVIDER_ID              | A unique identifier for flagd(grpc client) initiating the request.     | string                       | null                          | in-process              |
```

Comment:
`providerId`, `providerID`, or whatever fit the language best? We may not want to prescribe the names because the language may dictate (e.g. Python would be `provider_id`).

Reply:
For the gherkin tests in flagd_testbed we do have an implementation to transform camelcase to Kebapcase, which makes the testing easier, especially if we stick to one specific implementation. I would suggest to use `procividerId` as this will get automagically gets transformed to `provider_id` where as `providerID` might end up as `provider_i_d`.

I know we would need to change some implementations but I think for the ease of transformation, we should stick to a generic approach, which maps to all property name transformations for all kind of languages.

Reply:
See https://github.com/open-feature/python-sdk-contrib/blob/7048589b501f57dcd7ecd5a2cf6f2fcec55cde88/providers/openfeature-provider-flagd/tests/e2e/step/config_steps.py#L11

Reply:
I'm fine with that. FYI @toddbaert @cupofcat

Reply:
I am fine with using Id (camelcase) as long as we are consistent. Perhaps, for languages where it's more natural to have ID, we could provide an alias?